### ISSUE TITLE
In KDE Plasma Wayland with multiple vertically-stacked displays, the Flameshot GUI is shown only on the bottom display and shows content from the top display, truncating the rest of the viewport

### ISSUE THREAD

**Issue by AlexFolland** [flameshot-org/flameshot#3761]
### Flameshot Version

Flameshot 12.1.0

### Installation Type

Linux, MacOS, or Windows Package manager (apt, pacman, eopkg, choco, brew, ...)

### Operating System type and version

Manjaro

### Description

In KDE Plasma Wayland with multiple vertically-stacked displays, the Flameshot GUI is shown only on the bottom display and shows content from the top display, truncating the rest of the viewport.  The content from the bottom display is not fully shown, so screenshots can't be captured from there.  The top display is left unused.

Additionally, the following output is displayed in the terminal after running `flameshot gui`.

```Plaintext
Unable to get current screen, starting to use primary screen. It may be a cause of logical error and working with a wrong screen.
Unable to get current screen, starting to use primary screen. It may be a cause of logical error and working with a wrong screen.
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
Unable to get current screen, starting to use primary screen. It may be a cause of logical error and working with a wrong screen.
Unable to get current screen, starting to use primary screen. It may be a cause of logical error and working with a wrong screen.
flameshot: info: Screenshot aborted.
```

### Steps to reproduce

1. Have multiple vertically-stacked displays attached to the computer.
2. Run KDE Plasma Wayland.
3. Click the Flameshot system tray icon or run `flameshot gui` from a terminal.

### Screenshots or screen recordings

Here's a video made with my phone because I can't capture the issue clearly from the computer itself.  I transcoded the video to AV1 with CRF 59 to allow in-browser playback and fit the 10 MiB github upload limit, so if the video isn't playing for you, you'll need to enable AV1 video playback in your browser.

https://github.com/user-attachments/assets/df9b77af-2d1d-47de-9044-ab4f54b97756

### System Information

KDE Plasma Wayland

**Comment by FelixJochems** [comment]
This is because flameshot is unable to find what screen to capture, then spawns the screen capture on the screen where you're mouse currently is, squishing your bottom screen. 

You could set a window rule that forces flameshot to be fullscreen so that it will cover the entire screen that it's spawned on. Due to how flameshot is programmed right now it's hard to make it display on both screens at once, WLR layers could fix this in the future but is a big undertaking.

The easiest fix right now would be to install with the USE_WAYLAND_GRIM flag and set a windowrule to force flameshot into a fullscreen. This will start capture on the screen where your mouse currently is.

**Comment by AlexFolland** [comment]
> The easiest fix right now would be to install with the USE_WAYLAND_GRIM flag and set a windowrule to force flameshot into a fullscreen.

That would be a workaround, not a fix.  That being said, thanks for the workaround.  I'll try it soon and report whether it temporarily works around the bug.

**Comment by AlexFolland** [comment]
KWin doesn't seem to properly apply all window rules for this.  I realized that Flameshot's initial placement needs to be in the top-left corner of the total screen area, which is off any physical screen since my monitors have different resolutions and position (0,0) in the top-left corner is outside the screen area (in the black area in flameshot's preview, as can be seen in my video).  KWin needs to allow Flameshot to be placed there so that it's positioned perfectly, but I think KWin is preventing initial placement outside the physical screen space in the top-left corner.

**Comment by FelixJochems** [comment]
Ah yeah that is pretty dang annoying, I'm currently in the process of developing a fork targeted at wayland displays, especially because weird situations like these pop up way too often.

I'm not familiar with KWin/KDE Plasma so I don't know if there are other windowrules to force it somewhere else. The last option I can think of is compiling locally with the ```USE_WAYLAND_GRIM``` to see if that happens to change anything.

**Comment by typecasto** [comment]
I had a very similar issue with horizontal monitors, could you try setting up window rules as such and see if this fixes it?

![image](https://github.com/user-attachments/assets/50bafa66-97dc-4fe1-adba-909208149386)

**Comment by AlexFolland** [comment]
Yes, I tried that, but it doesn't work.  Screen position (0,0) is not on any display.

**Comment by AlexFolland** [comment]
The window appears with its top-left corner in the center of my lower display.

I've just tested an initial position of (320,0), and that does make it appear in the top-left corner of my top display, but shifted 320 pixels to the right, so I can see the black area that should be outside the display.  However, as soon as I set it to (319,0), it appears with its top-left corner in the middle of my lower display again.

**Comment by FelixJochems** [comment]
Definitely one of those edge cases I will keep in mind for the rewrite, but as it would require significant changes to the current codebase I will not be fixing this as is.

**Comment by AlexFolland** [comment]
It seems like this must not be Flameshot's fault.  I think I'll need to open a bug report for KWin and link to this ticket to clarify the issue for KWin devs.  What do you think?

**Comment by ParadoxLeon** [comment]
Maybe [this](https://github.com/flameshot-org/flameshot/issues/3686#issuecomment-2445400686) can help.

Had a similar Issue and this fixed it for me.
